File: v-lang/java.d/Spring/v-web/src/main/java/org/example/exception/BizException.java
package org.example.exception;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.http.HttpStatus;


@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class BizException extends RuntimeException {
    private String message;
    private HttpStatus httpStatus;

    public static BizException error() {
        return new BizException("服务器错误", HttpStatus.INTERNAL_SERVER_ERROR);
    }
}

File: v-lang/kt.d/v-spring/spring-web/src/main/kotlin/org/example/exception/BizException.kt
package org.example.exception

import org.springframework.http.HttpStatus

open class BizException(
    open val errorType: BizErrorTypeEnum = BizErrorTypeEnum.SERVER_ERROR,
    override val message: String = "服务器错误",
    open val httpStatus: HttpStatus = HttpStatus.INTERNAL_SERVER_ERROR
): RuntimeException()

enum class BizErrorTypeEnum {
    SERVER_ERROR,
    UNAUTHORIZED,
    UNREGISTERED,
    VERIFICATION
}